File: catalog/backends/node-express/patterns/slow_query_repository.js
const { Pool } = require('pg');

// Pattern: DBM slow query — artificial delay via pg_sleep
// Adapt: replace 'your_table' with a domain entity table
const pool = new Pool({
  connectionString: process.env.DATABASE_URL || 'postgresql://demo:demo@postgresql:5432/demo',
});

async function findAllSlow() {
  const { rows } = await pool.query(
    "SELECT *, pg_sleep(0.3) FROM your_table ORDER BY created_at DESC LIMIT 50"
  );
  return rows;
}

module.exports = { findAllSlow };

File: catalog/backends/ruby-rails/patterns/slow_query_repository.rb
# Pattern: DBM slow query — artificial delay via pg_sleep
# Adapt: replace 'your_table' with a domain entity table
DB_URL = ENV.fetch("DATABASE_URL", "postgresql://demo:demo@postgresql:5432/demo")

def self.find_all_slow
  sql = "SELECT *, pg_sleep(0.3) FROM your_table ORDER BY created_at DESC LIMIT 50"
  ActiveRecord::Base.connection.execute(sql).to_a
end
